### PR TITLE
Admin categories

### PR DESCRIPTION
**Pull request by dimahoncharenko**

### CODE REVIEW

File: src/middleware.ts [hunk 1]
```diff
@@ -28,12 +28,12 @@ export default auth((req) => {
     return null;
   }
 
-  if (isAuthRoute) {
-    if (isLoggedIn) {
-      return Response.redirect(new URL(DEFAULT_LOGIN_REDIRECT, nextUrl));
-    }
-    return null;
-  }
+  // if (isAuthRoute) {
```
Comment (SalOne22) [inline src/middleware.ts:31]: This comment must be reverted

File: src/modules/CategoriesTable/CategoriesTable.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          width={'100%'}
+          border={1}
+          borderColor='#EEE'
+          withTableBorder
+        >
+          <Table.Thead>{ths}</Table.Thead>
+          <Table.Tbody>{tbs}</Table.Tbody>
+        </Table>
+
+        {paginated.current.length > 1 && <PaginationBar
```
Comment (SalOne22) [inline src/modules/CategoriesTable/CategoriesTable.tsx:53]: For table pagination you can check how orders page implemented

File: src/shared/api/admin_categoryApi.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
```
Comment (SalOne22) [inline src/shared/api/admin_categoryApi.ts:1]: Why dont use existing api? Just add needed functions to it

**Review by SalOne22 — CHANGES_REQUESTED** [review]

**Comment by dimahoncharenko** [comment]
For now, each category doesn't support images, because firstly I need to get familiar a bit with Imgur API. And be aware, I added a mockery server for category api, so to make the table work you have to launch the server: npm run mockend

**Review by SalOne22 — APPROVED** [review]

**Pull request merged**